local_planner/src/nodes/common.cpp

```diff
@@ -115,6 +115,22 @@ void createPoseMsg(Eigen::Vector3f& out_waypt, Eigen::Quaternionf& out_q,
           Eigen::AngleAxisf(yaw, Eigen::Vector3f::UnitZ());
 }
 
+float getYawFromQuaternion(const Eigen::Quaternionf q){
```

Comment:
From wikipedia: 
```
	double siny_cosp = +2.0 * (q.w() * q.z() + q.x() * q.y());
	double cosy_cosp = +1.0 - 2.0 * (q.y() * q.y() + q.z() * q.z());  
	yaw = atan2(siny_cosp, cosy_cosp);
```

Reply:
I prefer doing that in a seperate PR. There are so many conventions for Quaternion and it took me some time to find this one (which actually works). I agree that we should find a more elegant way. But I think we also agreed to try and keep PRs as small and testable as possible